### PR TITLE
Update to newest XRTest API

### LINKED ISSUE #23634: Intermittent FAIL in /_mozilla/webxr/obtain_frame.html
```
  ▶ Unexpected subtest result in /_mozilla/webxr/obtain_frame.html:
  │ FAIL [expected PASS] obtain_frame
  │   → assert_approx_equals: left transform: Component number 12 should match expected 0.4 +/- 0.001 but got 0.5359500050544739
  │ 
  │ assert_matrix_approx_equals@http://web-platform.test:8000/_mozilla/webxr/resources/webxr-util.js:10:5
  │ @http://web-platform.test:8000/_mozilla/webxr/obtain_frame.html:32:17
  │ async*Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1594:20
  │ promise_test/tests.promise_tests</<@http://web-platform.test:8000/resources/testharness.js:591:31
  └ promise_test/tests.promise_tests<@http://web-platform.test:8000/resources/testharness.js:590:20
```

### PR DESCRIPTION
**Pull request by Manishearth**
requires https://github.com/servo/rust-webvr/pull/88 and https://github.com/servo/servo/pull/23575

Incorporates changes from https://github.com/immersive-web/webxr-test-api/pull/10 (and various minor improvements on that, like https://github.com/immersive-web/webxr-test-api/pull/20). Test no longer relies on racy state setting functions.

We're nearing consensus on what the test API should look like, thought I'd implement the updated model so I can start poking at the tests.

Fixes https://github.com/servo/servo/issues/23634 by setting everything at initialization.

This is based on https://github.com/servo/servo/pull/23575, so we can't merge it yet.

r? @asajeffrey (only the last commit)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23641)

### CODE REVIEW

File: components/script/dom/fakexrdevice.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
```
Comment (asajeffrey) [inline components/script/dom/fakexrdevice.rs]: Duplicate license.

File: components/script/dom/fakexrdevice.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            global,
+            FakeXRDeviceBinding::Wrap,
+        )
+    }
+
+    pub fn send_msg(&self, msg: MockVRControlMsg) {
+        self.global()
+            .as_window()
+            .webvr_thread()
+            .unwrap()
```
Comment (asajeffrey) [inline components/script/dom/fakexrdevice.rs]: Can we not panic here, especially since this is a public method?

**Review by asajeffrey — COMMENTED** [review]
Some minor niggles, otherwise r=me onceb you're ready to land.

File: components/script/dom/fakexrdevice.rs (hunk 2 above)
Comment (Manishearth) [reply components/script/dom/fakexrdevice.rs]: It's private now, and you shouldn't be running XR tests without the webvr thread existing.

**Comment by Manishearth** [comment]
@bors-servo r=asajeffrey

I made some small updates to get rid of cruft introduced by a lack of https://github.com/servo/servo/issues/23640

**Comment by CYBAI** [comment]
tidy complains 👀

```
$ ./mach test-tidy --no-progress --all
 0:07.50 INFO Diffing old and new manifests /home/travis/build/servo/servo/tests/wpt/mozilla/meta/MANIFEST.json
 0:07.63 WARNING Manifest /home/travis/build/servo/servo/tests/wpt/mozilla/meta/MANIFEST.json contains correct tests but file hashes changed.
 0:07.63 ERROR Manifest /home/travis/build/servo/servo/tests/wpt/mozilla/meta/MANIFEST.json is outdated, use |./mach update-manifest| to fix.
 0:07.63 INFO Diffing old and new manifests /home/travis/build/servo/servo/tests/wpt/metadata/MANIFEST.json
 0:12.38 INFO Diffing old and new manifests /home/travis/build/servo/servo/tests/wpt/webgl/meta/MANIFEST.json
Checking the config file...
Checking the wpt manifest file...
Checking directories for correct file extensions...
Running the WPT lint on ./tests/wpt/web-platform-tests/...
Running the WPT lint on ./tests/wpt/mozilla/tests/...
Checking files for tidiness...
./components/script/dom/fakexrdevice.rs:90: found an empty line following a {
Running the dependency licensing lint...
 85.7 MiB /  85.7 MiB (100 %)  15.0 MiB/s in  6s ETA:  0s
 61.9 MiB /  61.9 MiB (100 %)  18.5 MiB/s in  3s ETA:  0s
 85.7 MiB /  85.7 MiB (100 %)  10.7 MiB/s in  7s ETA:  0s
 61.9 MiB /  61.9 MiB (100 %)  14.1 MiB/s in  4s ETA:  0s
 11.4 MiB /  11.4 MiB (100 %)   4.9 MiB/s in  2s ETA:  0s
info: downloading component 'rustfmt'
info: installing component 'rustfmt'
Diff in /home/travis/build/servo/servo/components/script/dom/fakexrdevice.rs at line 87:
 }
 
 pub fn get_origin(origin: &FakeXRRigidTransformInit) -> Fallible<([f32; 3], [f32; 4])> {
-
     if origin.position.len() != 3 || origin.orientation.len() != 4 {
         return Err(Error::Type("Incorrectly sized array".into()));
     }
Run `./mach fmt` to fix the formatting
The command "./mach test-tidy --no-progress --all" exited with 1.
```

**Comment by Manishearth** [comment]
@bors-servo r=asajeffrey

Hmm, homu isn't catching this

**Comment by CYBAI** [comment]
```
{
    "status": "FAIL", 
    "group": "default", 
    "message": "Missing required member \"views\".", 
    "stack": "@http://web-platform.test:8000/_mozilla/webxr/create_session.html:9:33\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1594:25\nasync_test@http://web-platform.test:8000/resources/testharness.js:576:22\n@http://web-platform.test:8000/_mozilla/webxr/create_session.html:8:11\n", 
    "subtest": "create_session", 
    "test": "/_mozilla/webxr/create_session.html", 
    "line": 303554, 
    "action": "test_result", 
    "expected": "PASS"
}
```

**Comment by CYBAI** [comment]
```
{
    "status": "TIMEOUT", 
    "group": "default", 
    "message": "Test timed out", 
    "stack": null, 
    "subtest": "create_session", 
    "test": "/_mozilla/webxr/create_session.html", 
    "line": 305686, 
    "action": "test_result", 
    "expected": "PASS"
}
{
    "status": "ERROR", 
    "group": "default", 
    "message": "Incorrectly sized array", 
    "stack": "undefined:undefined:undefined", 
    "subtest": null, 
    "test": "/_mozilla/webxr/create_session.html", 
    "line": 305688, 
    "action": "test_result", 
    "expected": "OK"
}
```

**Comment by jdm** [comment]
```
+ ./mach test-tidy --no-progress --all
 0:05.29(B INFO(B Diffing old and new manifests /repo/tests/wpt/mozilla/meta/MANIFEST.json
 0:05.36(B WARNING(B Manifest /repo/tests/wpt/mozilla/meta/MANIFEST.json contains correct tests but file hashes changed.
 0:05.36(B ERROR(B Manifest /repo/tests/wpt/mozilla/meta/MANIFEST.json is outdated, use |./mach update-manifest| to fix.
 0:05.36(B INFO(B Diffing old and new manifests /repo/tests/wpt/webgl/meta/MANIFEST.json
 0:05.58(B INFO(B Diffing old and new manifests /repo/tests/wpt/metadata/MANIFEST.json
```

**Comment by Manishearth** [comment]
I ran that locally, it didn't do anything.

**Comment by Manishearth** [comment]
@bors-servo r=asajeffrey 

Deleted the file and it worked.

**Comment by Manishearth** [comment]
```
(view as text)

{
    "status": "FAIL", 
    "group": "default", 
    "message": "assert_equals: expected \"http://web-platform.test:8000/_mozilla/mozilla/xmlhttprequest_url.html\" but got \"http://web-platform.test:8000/resources/testharness.js\"", 
    "stack": "request.onload<@http://web-platform.test:8000/_mozilla/mozilla/xmlhttprequest_url.html:23:5\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1594:25\nTest.prototype.step_func_done/<@http://web-platform.test:8000/resources/testharness.js:1634:32\n", 
    "subtest": "Performance entries should contain the URL where the XMLHttpRequest originated", 
    "test": "/_mozilla/mozilla/xmlhttprequest_url.html", 
    "line": 254935, 
    "action": "test_result", 
    "expected": "PASS"
}
```

**Pull request merged**